host/sgx/sgxsign.c

```diff
@@ -640,6 +642,33 @@ static oe_result_t _init_sigstruct(
         mrenclave,
         sizeof(*mrenclave)));
 
+    if (attributes & SGX_FLAGS_KSS)
+    {
+        if (family_id)
+            OE_CHECK(oe_memcpy_s(
+                sigstruct->isvfamilyid,
+                sizeof(sigstruct->isvfamilyid),
+                family_id,
+                sizeof(*family_id)));
+        if (ext_product_id)
+            OE_CHECK(oe_memcpy_s(
+                sigstruct->isvextprodid,
+                sizeof(sigstruct->isvextprodid),
+                ext_product_id,
+                sizeof(*ext_product_id)));
+        sigstruct->attributemask.flags |= SGX_FLAGS_KSS;
+    }
+    else
+    {
+        sigstruct->attributemask.flags &= ~SGX_FLAGS_KSS;
```

Comment:
Do not unmask the `SGX_FLAGS_KSS` flag in the `attributemask` just because the enclave is not using KSS itself.

The `attributemask` determines whether two otherwise identical enclaves, one with and one without a particular attribute should be considered identical for attestation. Clearing this bit means that the dev wants them to be treated as the same (i.e. with or without the use of CONFIGID) Per [use case (3) in this comment](https://github.com/openenclave/openenclave/pull/3659#discussion_r508070776), this must be an explicit opt-in by the dev.

Reply:
Clearing the `SGX_FLAGS_KSS` in `attributemask` means the dev wants the loader to decide whether to enable/disable KSS feature (EINIT instruction will pass either way, as long as other policy is not violated, for example, non-zero family_id/extended_product_id requiring KSS enabled). For family_id/extedned_product_id feature only, it makes sense to keep `SGX_FALGS_KSS` in `attributemask` as 1. SGX remote attestation will reflect KSS enable/disable state in SGX `REPORT`/`QUOTE` and it's up to the Relying Party to decide whether the security property of the same enclave in KSS enabled and KSS disabled mode is equivalent if the dev allows the loader to choose KSS enable/disable.

Reply:
Will remove the unset/clearing. Thanks